internal/clients/interface.go

```diff
@@ -9,7 +9,7 @@ import (
 
 // GetSourcesClient returns Sources interface implementation. There are currently
 // two implementations available: HTTP and stub
-var GetSourcesClient func(ctx context.Context) (Sources, error)
+var GetSourcesClient func(ctx context.Context, url string) (Sources, error)
```

Comment:
This looks like an undocumented change. What is the purpose of this? I don't believe this belongs in here, you are forcing users of this interface to keep passing URLs around just because of a test.

I think it is worth looking into a way how to configure this via app configuration, problem is this is a global thing and tests can possibly run in parallel in the future.